### DISCUSSION TITLE
Returning prompts from expired cache may be disabled using a boolean argument

### DISCUSSION THREAD

**Discussion by ogunoz** [langfuse/langfuse#2746] [Ideas]
### Describe the feature or potential improvement

When we check the behavior in get_prompt function of PythonSDK, we realize that fallback is actually fallback's fallback many times because if in the past we manage to fill the cache for desired prompt, even though it is expired it will be returned. It is not really the behavior we would like to have. We may prefer fallback over expired cache so an optional boolean argument to disable returning from expired caches e.g., `allow_expired_cache_prompts: bool = True`

### Additional information

The requested feature is for Python SDK

**Comment by marcklingen** [comment]
Thanks for sharing! With this change you might increase the frequency of background refreshes in order to have a more "fresh" prompt without sacrificing performance. Is this a solution for you or why is fallback better than slightly stale in your case?

https://langfuse.com/changelog/2024-09-04-prompt-management-zero-latency
